### PR TITLE
Also desugar with-do

### PR DESCRIPTION
**Pull request by b-studios**

### CODE REVIEW

File: effekt/shared/src/main/scala/effekt/RecursiveDescent.scala [hunk 1]
```diff
@@ -221,6 +221,8 @@ class RecursiveDescent(positions: Positions, tokens: Seq[Token], source: Source)
      case Var(id) =>
        val tgt = IdTarget(id)
        Return(Call(tgt, Nil, Nil, (BlockLiteral(Nil, vparams, bparams, body)) :: Nil))
+     case Do(effect, id, targs, vargs, bargs) =>
+      Return(Do(effect, id, targs, vargs, (BlockLiteral(Nil, vparams, bparams, body)) :: Nil))
```
Comment (phischu) [inline effekt/shared/src/main/scala/effekt/RecursiveDescent.scala:225]: You are dropping `bargs`, why not (like in the other cases `bargs :+ (BlockLiteral(Nil, vparams, bparams, body))`?

Comment (b-studios) [reply effekt/shared/src/main/scala/effekt/RecursiveDescent.scala:225]: Good point!

Comment (b-studios) [inline effekt/shared/src/main/scala/effekt/RecursiveDescent.scala:225]:
```suggestion
      Return(Do(effect, id, targs, vargs, bargs :+ (BlockLiteral(Nil, vparams, bparams, body)) :: Nil))
```

File: effekt/shared/src/main/scala/effekt/RecursiveDescent.scala [hunk 2]
```diff
@@ -221,6 +221,8 @@ class RecursiveDescent(positions: Positions, tokens: Seq[Token], source: Source)
      case Var(id) =>
        val tgt = IdTarget(id)
        Return(Call(tgt, Nil, Nil, (BlockLiteral(Nil, vparams, bparams, body)) :: Nil))
+     case Do(effect, id, targs, vargs, bargs) =>
+      Return(Do(effect, id, targs, vargs, bargs :+ (BlockLiteral(Nil, vparams, bparams, body)) :: Nil))
```
Comment (b-studios) [inline effekt/shared/src/main/scala/effekt/RecursiveDescent.scala:225]:
```suggestion
      Return(Do(effect, id, targs, vargs, bargs :+ BlockLiteral(Nil, vparams, bparams, body)))
```

**Review by phischu — APPROVED** [review]

**Pull request merged**